beacon-chain/sync/metrics.go

```diff
@@ -64,29 +79,36 @@ func (s *Service) updateMetrics() {
 		return
 	}
 	// We update the dynamic subnet topics.
-	digest, err := s.forkDigest()
+	digest, err := s.currentForkDigest()
 	if err != nil {
 		log.WithError(err).Debugf("Could not compute fork digest")
 	}
 	indices := s.aggregatorSubnetIndices(s.cfg.Chain.CurrentSlot())
+	syncIndices := cache.SyncSubnetIDs.GetAllSubnets(core.SlotToEpoch(s.cfg.Chain.CurrentSlot()))
 	attTopic := p2p.GossipTypeMapping[reflect.TypeOf(&pb.Attestation{})]
+	syncTopic := p2p.GossipTypeMapping[reflect.TypeOf(&pb.SyncCommitteeMessage{})]
 	attTopic += s.cfg.P2P.Encoding().ProtocolSuffix()
+	syncTopic += s.cfg.P2P.Encoding().ProtocolSuffix()
 	if flags.Get().SubscribeToAllSubnets {
 		for i := uint64(0); i < params.BeaconNetworkConfig().AttestationSubnetCount; i++ {
-			formattedTopic := fmt.Sprintf(attTopic, digest, i)
-			topicPeerCount.WithLabelValues(formattedTopic).Set(float64(len(s.cfg.P2P.PubSub().ListPeers(formattedTopic))))
+			s.collectMetricForSubnet(attTopic, digest, i)
+		}
+		for i := uint64(0); i < params.BeaconConfig().SyncCommitteeSubnetCount; i++ {
+			s.collectMetricForSubnet(syncTopic, digest, i)
 		}
 	} else {
 		for _, committeeIdx := range indices {
-			formattedTopic := fmt.Sprintf(attTopic, digest, committeeIdx)
-			topicPeerCount.WithLabelValues(formattedTopic).Set(float64(len(s.cfg.P2P.PubSub().ListPeers(formattedTopic))))
+			s.collectMetricForSubnet(attTopic, digest, committeeIdx)
+		}
+		for _, committeeIdx := range syncIndices {
+			s.collectMetricForSubnet(syncTopic, digest, committeeIdx)
 		}
 	}
 
 	// We update all other gossip topics.
 	for _, topic := range p2p.AllTopics() {
 		// We already updated attestation subnet topics.
-		if strings.Contains(topic, "beacon_attestation") {
+		if strings.Contains(topic, p2p.GossipAttestationMessage) || strings.Contains(topic, p2p.GossipSyncCommitteeMessage) {
```

Comment:
These strings contains across the package could be simplified via an abstraction for checking topic kinds. Ideally the implementation details of how topic containment is actually checked is not important to this function, and can make for cleaner code. Something to keep in mind for future cleanup

Reply:
Yeap, this has been brought up by kasey earlier too. We have an issue to track it here
#9480 , but it will take quite a bit of re-jigging stuff so I haven't really looked into it.